### DISCUSSION TITLE
Can the config for each task of a process be dynamically changed according to the type of samples?

### DISCUSSION THREAD

**Discussion by zhangzhen** [nextflow-io/nextflow#2343] [Q&A] [answered]
Say we have a process, and 4 samples need to be run on the process simultaneously. Sample 1 and 2 have the type A, and Sample 3 and 4 have the type B. For samples of type A one command needs to be run, and for samples of type b the other command needs to be run. Can one just use the config file to set this at one time rather than manually modify it in each process? 

Thanks,
Zhen

**Comment by manuelesimi** [comment, marked as answer]
Hello,

I'd say it depends on some conditions. For instance, do you have only A and B, or at the next execution would also have C and D?

Regardless, assuming the information on the type is available in a channel, options I would consider:

1. use [conditional scripts](https://www.nextflow.io/docs/latest/process.html#conditional-scripts) in your process to add/remove/skip commands according to the type, or
2. [branch](https://www.nextflow.io/docs/latest/operator.html#branch) the channel according to the type of samples and then have 2 processes, one for each type (again, it depends on how many types you expect and how complex is the process to split)

I'm not sure that using only the config file to achieve is doable because I think you'd still need to add configurable conditions to the process. 

Thanks,
manuele
